riptide-stream/src/main/java/org/zalando/riptide/stream/StreamSpliterator.java

```diff
@@ -86,4 +86,13 @@ public long estimateSize() {
     public int characteristics() {
         return ORDERED | IMMUTABLE;
     }
+
+    @Override
+    public void close() throws UncheckedIOException {
```

Comment:
I do not think this is a problem at all.

The input stream will be closed whenever the connection is closed. This is part of the lower layer protocols. As long as we take care that we close the stream/connection, when some error happens in our receiver, we will have no stale resources hanging.

As far as I see, this is taken care of by the current implementation.